server/java/src/main/java/com/styra/tickethub/TicketHub.java

```diff
@@ -98,6 +137,46 @@ static Map<String, Object> getSessionAttributes(HttpServletRequest request) {
                 }).orElse(Map.of());
     }
 
+    private String getSessionPath() {
+        return getSessionPath(request);
+    }
+
+    static String getSessionPath(HttpServletRequest request) {
+        return request.getPathInfo();
+    }
+
+    private String getSessionMethod() {
+        return getSessionMethod(request);
+    }
+
+    static String getSessionMethod(HttpServletRequest request) {
+        return request.getMethod();
+    }
+
+    private boolean authz() {
+        System.out.printf("DEBUG: %s %s %s\n", getSessionPath(), getSessionMethod(), getSessionAttributes());
+
+        java.util.Map<String, Object> iMap = java.util.Map.ofEntries(
+            entry("path", getSessionPath()),
+            entry("method", getSessionMethod()),
+            entry("cookie", getSessionAttributes()),
+            entry("user", getSessionAttributes().get("subject")),
+            entry("tenant", getSessionAttributes().get("tenant")),
+            entry("action", getSessionMethod().toLowerCase())
```

Comment:
This is OK for now, I suppose, but I think it would be nice if we had one layer of abstraction more in here: Instead of feeding HTTP-related things as input, we'd translate them into domain-related things. Like, `GET /tickets` wouldn't become
```json
{
  "path": "/tickets",
  "action": "get"
}
```
but rather something like
```json
{
  "resource": "tickets",
  "action": "list"
}
```
or something like that. Rationale is that this gets us further away from the API gateway approach, to demonstrate the value of doing this in the application, not outside of it.

Reply:
Hmm, yeah that's an interesting idea. I think it might be interesting to explore having a way to use Rego to translate low-level HTTP call information into a more abstracted "application layer" representation. Of course that's not the only way you could do it. But that's out of scope for this PR IMO.